### ISSUE TITLE
Layout rerender every route navigate after build but dev is fine

### ISSUE THREAD

**Issue by zoom2009** [vercel/next.js#63040]
### Link to the code that reproduces this issue

https://github.com/zoom2009/nextjs-layout-rerender

### To Reproduce

**Develop case (Working fine)**
1. yarn dev
2. open inspect and press info or home to see a log

**Production case (bug by call useEffect every route navigate)**
1. yarn build
2. yarn start
3. open inspect and press info or home to see a log

### Current vs. Expected behavior

**Current**
after build useEffect call every route navigate that make my layout re-render every time how can I shallow it?

**Expect**
like `yarn dev` useEffect call only once in layout

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:52 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T8103
Binaries:
  Node: 20.10.0
  npm: 10.4.0
  Yarn: 1.22.21
  pnpm: N/A
Relevant Packages:
  next: 14.1.3
  eslint-config-next: 14.1.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router

### Which stage(s) are affected? (Select all that apply)

next build (local), Vercel (Deployed)

### Additional context

**develop mode `yarn dev` is work well**

https://github.com/vercel/next.js/assets/33366155/1ad448ef-58ee-4300-84de-577e6857fa9a

<hr />

**production mode `yarn build && yarn start` is re-render every route navigate**

https://github.com/vercel/next.js/assets/33366155/d5d58dd3-e592-4bfb-bc6c-1da6ed65d3ec

<hr />

**Here my reproduce code**
https://github.com/zoom2009/nextjs-layout-rerender

Do you guy have any solution for do not make layout re-render every route navigate ?
thank you.

**Comment by coffeecupjapan** [comment]
It is weird if you remove `memo` from code, it works fine in production mode.

**Comment by zoom2009** [comment]
> It is weird if you remove `memo` from code, it works fine in production mode.

it's work but very weird 😞 by the way thank you for your work around.
I hope next will fix this.

**Comment by zoom2009** [comment]
> Possibly related to / duplicate of [#52558](https://github.com/vercel/next.js/issues/52558).

It not related dude

**Comment by marcoshernanz** [comment]
Thanks for the report. This is the same App Router root layout rerendering issue tracked in #52558. Client navigation recreates layout state in production even though the root layout should remain mounted. Closing as a duplicate so the rendering investigation stays consolidated.
